test/features/caching/mailbox_cache_client_test.dart

```diff
@@ -0,0 +1,181 @@
+import 'dart:io';
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:tmail_ui_user/features/caching/config/hive_cache_config.dart';
+import 'package:tmail_ui_user/features/caching/mailbox_cache_client.dart';
+import 'package:tmail_ui_user/features/mailbox/data/extensions/mailbox_extension.dart';
+
+import '../../fixtures/mailbox_fixtures.dart';
+
+void main() {
+
+  late MailboxCacheClient _mailboxCacheClient;
+
+  setUpAll(() {
+    HiveCacheConfig().setUp(cachePath: Directory.current.path);
+  });
+
+  setUp(() {
+    _mailboxCacheClient = MailboxCacheClient();
+  });
+
+  group('[delete]', () {
+    test('cache should delete item successfully when cache empty', () async {
+      await _mailboxCacheClient.deleteItem(MailboxFixtures.inboxMailbox.id.toString());
+
+      final remainingItems = await _mailboxCacheClient.getAll();
+
+      expect(remainingItems.length, equals(0));
+    });
+
+    test('cache should not delete item which not in the list', () async {
+      await _mailboxCacheClient.insertItem(
+          MailboxFixtures.inboxMailbox.id.toString(),
+          MailboxFixtures.inboxMailbox.toMailboxCache());
+
+      await _mailboxCacheClient.insertItem(
+          MailboxFixtures.sentMailbox.id.toString(),
+          MailboxFixtures.sentMailbox.toMailboxCache());
+
+      await _mailboxCacheClient.deleteItem(MailboxFixtures.folder1.id.toString());
+
+      final remainingItems = await _mailboxCacheClient.getAll();
+
+      expect(remainingItems.length, equals(2));
+      expect(
+        remainingItems,
+        containsAll({
+          MailboxFixtures.inboxMailbox.toMailboxCache(),
+          MailboxFixtures.sentMailbox.toMailboxCache()
+        }));
+    });
+
+    test('cache should delete item successfully', () async {
+      await _mailboxCacheClient.insertItem(
+        MailboxFixtures.inboxMailbox.id.toString(),
+        MailboxFixtures.inboxMailbox.toMailboxCache());
+
+      await _mailboxCacheClient.insertItem(
+        MailboxFixtures.sentMailbox.id.toString(),
+        MailboxFixtures.sentMailbox.toMailboxCache());
+
+      await _mailboxCacheClient.deleteItem(MailboxFixtures.inboxMailbox.id.toString());
+
+      final remainingItems = await _mailboxCacheClient.getAll();
+
+      expect(remainingItems.length, equals(1));
+      expect(remainingItems[0], equals(MailboxFixtures.sentMailbox.toMailboxCache()));
```

Comment:
IMO It is advisable to check that `remainingItems` still contains an `inboxMailbox` that is not correct in the meaning of this case.